File: javascript/complete/no001.js
#!/usr/bin/env node

/* If we list all the natural numbers below 10 that are multiples of 3 or 5,
   we get 3, 5, 6 and 9. The sum of these multiples is 23.

   Find the sum of all the multiples of 3 or 5 below 1000. */

var operator = require('../operator.js'),
    timer = require('../timer.js');

function filterHelper(values) {
  var result = function(n) {
    for(var i = 0, val; value = values[i]; i++) {
      if (!(n % value)) {
        return true;
      }
    }
    return false;
  };

  return result;
};

exports.main = function(verbose) {
  if (typeof verbose == 'undefined') {
    verbose = false;
  }
  var multiples = operator.range(1, 1000);
  multiples = multiples.filter(filterHelper([3, 5]));

  return operator.sum(multiples);
};

if (require.main === module) {
  timer.timer(1, exports.main);
}

File: python/complete/no001.py
#!/usr/bin/env python

# Find the sum of all the multiples of 3 or 5 below 1000.

from python.decorators import euler_timer

def main(verbose=False):
    return sum(i for i in range(1, 1000) if i % 3 == 0 or i % 5 == 0)

if __name__ == '__main__':
    print euler_timer(1)(main)(verbose=True)
